File: app/src/main/java/com/baronzhang/android/weather/di/component/PresenterComponent.java
package com.baronzhang.android.weather.di.component;

import com.baronzhang.android.weather.di.module.ApplicationModule;
import com.baronzhang.android.weather.view.home.drawer.DrawerMenuPresenter;
import com.baronzhang.android.weather.view.selectcity.SelectCityPresenter;

import javax.inject.Singleton;

import dagger.Component;
import com.baronzhang.android.weather.view.home.HomePagePresenter;

/**
 * @author 张磊 (baron[dot]zhanglei[at]gmail[dot]com)
 *         2016/12/2
 */
@Singleton
@Component(modules = {ApplicationModule.class})
public interface PresenterComponent {

    void inject(HomePagePresenter presenter);

    void inject(SelectCityPresenter presenter);

    void inject(DrawerMenuPresenter presenter);
}
 

File: weather-mvp-dagger-kotlin/src/main/java/com/minimalist/weather/kotlin/di/PresenterComponent.kt
package com.minimalist.weather.kotlin.di


import com.minimalist.weather.kotlin.view.home.HomePagePresenter
import com.minimalist.weather.kotlin.view.home.drawer.DrawerMenuPresenter
import com.minimalist.weather.kotlin.view.selectcity.SelectCityPresenter
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface PresenterComponent {

    fun inject(presenter: HomePagePresenter)

    fun inject(presenter: SelectCityPresenter)

    fun inject(presenter: DrawerMenuPresenter)
}
 